pkg/resolver/search.go

```diff
@@ -265,321 +276,69 @@ func (s *SearchResult) resolveItems() ([]map[string]interface{}, error) {
-				then = now.AddDate(0, -1, 0).Format(format)
-
-			case "year":
-				then = now.AddDate(-1, 0, 0).Format(format)
-
-			default:
-				//check that property value is an array:
-				array := strings.Split(string(val), ":")
-				if len(array) > 1 {
-					klog.V(7).Info("filter is array. Operator is @>.")
-					operator = "@>"
-
-				} else {
-					if _, ok := arrayProperties[filter]; ok {
-						klog.V(7).Info("filter ", filter, " is present in arrayProperties. Using operator ?|.")
-						operator = "?|"
-					} else {
-						klog.V(7).Info("filter is neither label nor in arrayProperties: ", filter)
-						operator = ""
-					}
-				}
-				then = val
-			}
-			// Add the value and operator to map
-			updateOpValueMap(operator, opValueMap, then)
-		}
-	}
-	return opValueMap
-}
-
-// Labels are sorted alphabetically to ensure consistency, then encoded in a
-// string with the following format.
-// key1:value1; key2:value2; ...
-func formatLabels(labels map[string]interface{}) string {
-	keys := make([]string, 0)
-	labelStrings := make([]string, 0)
-	for k := range labels {
-		keys = append(keys, k)
-	}
-	sort.Strings(keys)
-	for _, k := range keys {
-		labelStrings = append(labelStrings, fmt.Sprintf("%s=%s", k, labels[k]))
-	}
-	return strings.Join(labelStrings, "; ")
-}
-
-// Encode array into a single string with the format.
-//  value1; value2; ...
-func formatArray(itemlist []interface{}) string {
-	keys := make([]string, len(itemlist))
-	for i, k := range itemlist {
-		keys[i] = convertToString(k)
-	}
-	sort.Strings(keys)
-	return strings.Join(keys, "; ")
-}
-
-// Convert interface to string format
-func convertToString(data interface{}) string {
-	var item string
-	switch v := data.(type) {
-	case string:
-		item = strings.ToLower(v)
-	case bool:
-		item = strconv.FormatBool(v)
-	case float64:
-		item = strconv.FormatInt(int64(v), 10)
-	default:
-		klog.Warningf("Error formatting property with type: %+v\n", reflect.TypeOf(v))
-	}
-	return item
-}
-
-func formatDataMap(data map[string]interface{}) map[string]interface{} {
-	item := make(map[string]interface{})
-	for key, value := range data {
-		switch v := value.(type) {
-		case string:
-			item[key] = v //strings.ToLower(v)
-		case bool:
-			item[key] = strconv.FormatBool(v)
-		case float64:
-			item[key] = strconv.FormatInt(int64(v), 10)
-		case map[string]interface{}:
-			item[key] = formatLabels(v)
-		case []interface{}:
-			item[key] = formatArray(v)
-		default:
-			klog.Warningf("Error formatting property with key: %+v  type: %+v\n", key, reflect.TypeOf(v))
-			continue
-		}
-	}
-	return item
-}
-
-// helper function to point values in string  array
-func pointerToStringArray(pointerArray []*string) []string {
-
-	values := make([]string, len(pointerArray))
-	for i, val := range pointerArray {
-		values[i] = *val
-	}
-	return values
-}
+func WhereClauseFilter(ctx context.Context, input *model.SearchInput,
+	propTypeMap map[string]string) ([]exp.Expression, map[string]string, error) {
```

Comment:
Is this better?
```suggestion
func (s *SearchResult) WhereClauseFilter(ctx context.Context, input *model.SearchInput) ([]exp.Expression, error) {
```

Reply:
Also `context` is in the SearchResult struct.
```suggestion
func WhereClauseFilter(input *model.SearchInput,
	propTypeMap map[string]string) ([]exp.Expression, map[string]string, error) {
	
	// access with 
	// s.propTypeMap
	// s.context
```

Reply:
I'm not sure this would work, because we use this where clause for search complete too.

Reply:
ok, thanks for clarifying.